### ISSUE TITLE
unexpected PushConstantOutOfRange with buffer_reference

### ISSUE THREAD

**Issue by pixeljetstream** [KhronosGroup/Vulkan-ValidationLayers#2467]
**Describe the Issue**
Following setup should be correct behavior, but throws the errors below on pipeline creation.

```
// pipelineLayout
    VkPushConstantRange pushRanges[2];
    pushRanges[0].stageFlags = VK_SHADER_STAGE_VERTEX_BIT;
    pushRanges[0].size       = sizeof(uint64_t);
    pushRanges[0].offset     = 0;
    pushRanges[1].stageFlags = VK_SHADER_STAGE_FRAGMENT_BIT;
    pushRanges[1].size       = sizeof(uint64_t);
    pushRanges[1].offset     = sizeof(uint64_t);

// VertexShader
layout(buffer_reference, buffer_reference_align=16, scalar) readonly buffer MatrixBuffer {
  MatrixData  matrix;
};

layout(push_constant, scalar) uniform pushConstants {
  MatrixBuffer v;
};

// Fragment Shader
layout(buffer_reference, buffer_reference_align=16, scalar) readonly buffer MaterialBuffer {
  MaterialData  material;
};
  
layout(push_constant, scalar) uniform pushConstants {
  layout(offset=8)
  MaterialBuffer v;
};
```
**Valid Usage ID**
ERROR: UNASSIGNED-CoreValidation-Shader-PushConstantOutOfRange
 --> Validation Error: [ UNASSIGNED-CoreValidation-Shader-PushConstantOutOfRange ] Object 0: handle = 0xecc2050000000040, type = VK_OBJECT_TYPE_SHADER_MODULE; Object 1: handle = 0x203fc60000000023, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x650ba96a | Push-constant buffer: member:0 member:0 member:0 index:[0][2] offset:0 in VK_SHADER_STAGE_VERTEX_BIT is out of range in VkPipelineLayout 0x203fc60000000023[].

ERROR: UNASSIGNED-CoreValidation-Shader-PushConstantOutOfRange
 --> Validation Error: [ UNASSIGNED-CoreValidation-Shader-PushConstantOutOfRange ] Object 0: handle = 0xfd77e40000000041, type = VK_OBJECT_TYPE_SHADER_MODULE; Object 1: handle = 0x203fc60000000023, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x650ba96a | Push-constant buffer: member:0 member:0 member:0 index:[0] member:0 index:[2] offset:0 in VK_SHADER_STAGE_FRAGMENT_BIT is out of range in VkPipelineLayout 0x203fc60000000023[].

**Environment:**
 - OS: Windows 10
 - GPU: GeForce RTX 3080
 - SDK or header version if building from repo: 1.2.162.0
 - Options enabled (synchronization, best practices, etc.): 

**Additional context**

**Comment by pixeljetstream** [comment]
https://drive.google.com/file/d/1Dalx_9WyPchZcROp95ISAl6yTS76_I6J/view?usp=sharing reproducer program, let me know when you have it so I can delete it

**Comment by pixeljetstream** [comment]
while encountered in a different app, a colleague mentioned that sdk 148 didn't have that problem but he sees same error in 162

**Comment by mark-lunarg** [comment]
@ncesario-lunarg, can you grab the repo app (for later) and let Christophe know?  Or I can grab it, either way.

**Comment by ncesario-lunarg** [comment]
@pixeljetstream @mark-lunarg got it and the app runs on my machine. I will take a look at this later today.

**Comment by ncesario-lunarg** [comment]
Closing as this should be resolved by #2497. @pixeljetstream I am not seeing this locally with the `vk_pushconstant_address.exe` you referenced, but please re-open if you are still encountering this problem.
